### DISCUSSION TITLE
There is a way to stop the diagnostics?

### DISCUSSION THREAD

**Discussion by KamiC6238** [TypeFox/monaco-languageclient#471] [Q&A]
Some users on our website only want to use code autocompletion, but not code error diagnosis. Does monaco-languageclient provide a way to turn on/off code error diagnosis?

**Comment by CGNonofr** [comment]
Some language server implementations have a configuration that allows to disable it, what language server do you use?

Otherwise, the diagnostic client feature can be disabled but it's not easy with the current state of the code

**Reply by KamiC6238** [reply]
> Some language server implementations have a configuration that allows to disable it, what language server do you use?
> 
> Otherwise, the diagnostic client feature can be disabled but it's not easy with the current state of the code

Thanks for your answer. Finally i tried to rewrite the `socket.onMessage` like that:
```javascript
const socket = toSocket(this.websocket);

socket.onMessage = callback => {
    if (this.websocket) {
        this.websocket.onmessage = event => {
            if (
                typeof event.data === 'string' &&
                event.data.includes('textDocument/publishDiagnostic')
            ) {
                if (!this.enableDiagnostics) {
                    return;
                }
                callback(event.data);
            } else {
                callback(event.data);
            }
        };
    }
}
```

And it works for me.

**Reply by CGNonofr** [reply]
Ok It's kind of a hack but If you're ok with it then it's fine
